R/tidytlg-package.R

```diff
@@ -101,18 +101,55 @@
 #' }
 #'
 #'
-#' @importFrom magrittr %>%
-#' @importFrom rlang .data
 #'
-#' @docType package
 #' @name tidytlg
 #' @keywords internal
 "_PACKAGE"
 
-globalVariables(c("anbr", "label", "rowvar", "rowcondition", "row_type",
+utils::globalVariables(c("anbr", "label", "rowvar", "rowcondition", "row_type",
                   "rowtext", "tableby", "by", "func", "colvar", "header_order",
                   "df", "filter_to_df", "statlist", "denom_df", "precision",
-                  "tbltype", "dotdotdot", ":="))
+                  "tbltype", "dotdotdot",
+                  ".", "name", "alpha", "value", "N", "pct", "denom", "results",
+                  "n_ord", "n_ord2", "n_ord3", "data_nest", "nested_treatment",
+                  "nested_level", "nest_treatment", "stat", "extra", "prec",
+                  "IDENTIFIER", "TABLE ID", "decode", "colnbr", "newrows", "el",
+                  "sorter", "coldef", "V1"))
+
+## usethis namespace: start
+#' @importFrom rlang enexprs is_call expr sym %||% exprs syms as_string is_named
+#'   eval_tidy enexpr as_name .data :=
+#' @importFrom tidyr pivot_wider pivot_longer complete nesting nest unnest
+#'   replace_na crossing fill
+#' @importFrom dplyr select bind_rows ungroup mutate rowwise filter summarise
+#'   bind_cols
+#' @importFrom dplyr group_by distinct n left_join rename relocate across
+#'   if_else
+#' @importFrom dplyr arrange case_when desc cur_group_id inner_join rename_with
+#'   row_number
+#' @importFrom dplyr group_by_at slice mutate_if
+#' @importFrom stringr str_detect str_remove str_replace str_split regex
+#'   str_to_lower
+#' @importFrom dplyr all_of any_of everything starts_with ends_with
+#' @importFrom forcats fct_expand fct_recode fct_relevel
+#' @importFrom magrittr %>% extract2
+#' @importFrom purrr map walk map_dfr pmap discard compact map_lgl imap map_chr
+#' @importFrom tibble tibble tribble add_row tibble_row
+#' @importFrom methods formalArgs
+#' @importFrom stats setNames lag
+#' @importFrom utils head modifyList tail
+#' @importFrom huxtable everywhere
+#' @importFrom crayon bold underline
+#' @importFrom readxl read_excel
+#' @importFrom cellranger cell_cols
+#' @importFrom png readPNG
+#' @importFrom ggplot2 ggsave
+#' @importFrom grDevices dev.cur dev.off pdf
+#' @importFrom rstudioapi getSourceEditorContext isAvailable
+## usethis namespace: end
+NULL
```

Comment:
I am not familiar with `usethis` setup and configuration. Do you mind explaining to me the reason this has to be here?

Reply:
it is just a matter of keeping everything in the same place. 

If you call `use_import_from("rlang", ".data")`, it will add it there. the comment is just to let it know where to put it. 


there was some `#' @import`, so I just moved it to a single place

Reply:
I understand. Looks like it's a decision that has been made at some point to gather these together.

T]I generally recommend against using @importFrom for a couple of reasons:
* readability sinks when the function calls are not prepended by their package name
* `@importFrom` actually copies the reference to the function to the package namespace, so it is accessible and visible by `tidytlg::<function_name>` which is annoying to deal with in terms of keeping the scope of the public API clear to the end-users, which I think is especially important if these end-users are not veteran R programmers.
* rarely, it causes name clashes

For the above reasons, I would argue against `@importFrom` as a rule of thumb (with occasional exceptions for often-used operators, like magrittr::`%>%`). I might make an effort to remove this in the future. For now let it stay.

Reply:
I agree! It is just that some were in zzz.R while others were in tidytlg-package.R. In the new code I wrote, I used namespaced only! 

This change should be tackled in other PRs indeed.